go/2_renew-token.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"fmt"
+	"log"
+
+	vault "github.com/hashicorp/vault/api"
+)
+
+// Once you've set the token for your Vault client, you will need to periodically renew its lease.
+// A function like this should be run as a goroutine to avoid blocking.
+func renewToken(client *vault.Client) {
+	for {
+		vaultLoginResp, err := login(client)
+		if err != nil {
+			log.Fatalf("unable to authenticate to Vault: %v", err)
+		}
+		tokenErr := manageTokenLifecycle(client, vaultLoginResp)
+		if tokenErr != nil {
+			log.Fatalf("unable to start managing token lifecycle: %v", tokenErr)
+		}
+	}
+}
+
+// Starts token lifecycle management. Returns only fatal errors as errors, otherwise returns nil so we can attempt login again.
+func manageTokenLifecycle(client *vault.Client, token *vault.Secret) error {
+	renew := token.Auth.Renewable // You may notice a different top-level field called Renewable. That one is used for dynamic secrets renewal, not token renewal.
+	if renew {
```

Comment:
It's more idiomatic to do early returns, i.e. if !renew { log, return nil }
Makes it a bit more readable because of less indents.

Reply:
That looks so much better, you're right!